pkg/operation/botanist/waiter.go

```diff
@@ -369,3 +373,20 @@ func (b *Botanist) WaitUntilNodesDeleted(ctx context.Context) error {
 		return retry.MinorError(fmt.Errorf("not all nodes have been deleted in the shoot cluster"))
 	})
 }
+
+// WaitUntilManagedResourcesDeleted waits until all managed resources are gone or the context is cancelled.
+func (b *Botanist) WaitUntilManagedResourcesDeleted(ctx context.Context) error {
+	return retry.Until(ctx, 5*time.Second, func(ctx context.Context) (done bool, err error) {
+		managedResources := &resourcesv1alpha1.ManagedResourceList{}
+		if err := b.K8sSeedClient.Client().List(ctx, managedResources, client.InNamespace(b.Shoot.SeedNamespace)); err != nil {
+			return retry.SevereError(err)
+		}
+
+		if len(managedResources.Items) == 0 {
+			return retry.Ok()
+		}
+
+		b.Logger.Infof("Waiting until all managed resources have been deleted in the shoot cluster...")
+		return retry.MinorError(fmt.Errorf("not all managed resources have been deleted in the shoot cluster"))
```

Comment:
In the error message here, can you provide the names of the managed resources that are still left?